src/lib.rs

```diff
@@ -267,77 +302,184 @@ fn interface_and_mtu_windows(socket: &UdpSocket) -> Result<(String, usize), Erro
+    fn socket_with_addr(local_ip: IpAddr) -> SocketAddr {
+        loop {
+            let port = rand::thread_rng().gen_range(1024..65535);
+            let socket = UdpSocket::bind(SocketAddr::new(local_ip, port));
+            match socket {
+                Ok(socket) => {
+                    if let Ok(saddr) = socket.local_addr() {
+                        // We found an unused port.
+                        return saddr;
+                    }
                 }
+                Err(e) => match e.kind() {
+                    std::io::ErrorKind::AddrInUse => {
+                        // We hit a used port, try again.
+                        continue;
+                    }
+                    _ => {
+                        // We hit another error. Pretend that worked by returning the socket
+                        // address, so the actual code can hit the same error.
+                        return SocketAddr::new(local_ip, port);
+                    }
+                },
             }
-        } else {
-            // Some GitHub runners don't have IPv6. Just warn if we can't get an IPv6 address.
-            assert!(!ipv4);
-            warn!("No IPv6 address found for {sockaddr}");
         }
     }
 
+    fn local_v4() -> SocketAddr {
+        socket_with_addr(IpAddr::V4(Ipv4Addr::LOCALHOST))
+    }
+
+    fn local_v6() -> SocketAddr {
+        socket_with_addr(IpAddr::V6(Ipv6Addr::LOCALHOST))
+    }
+
+    fn inet_v4() -> SocketAddr {
+        // cloudflare.com
+        socket_with_addr(IpAddr::V4(Ipv4Addr::new(104, 16, 132, 229)))
+    }
+
+    fn inet_v6() -> SocketAddr {
+        // cloudflare.com
+        socket_with_addr(IpAddr::V6(Ipv6Addr::new(
+            0x26, 0x06, 0x47, 0x00, 0x68, 0x10, 0x84, 0xe5,
+        )))
+    }
+
     #[test]
-    fn loopback_interface_mtu_v4() {
-        #[cfg(target_os = "macos")]
-        check_mtu("localhost:443", true, 16384);
-        #[cfg(target_os = "linux")]
-        check_mtu("localhost:443", false, 65_536);
-        #[cfg(target_os = "windows")]
-        check_mtu("localhost:443", false, 4_294_967_295);
+    fn loopback_v4_loopback_v4() {
+        assert!(interface_and_mtu(&(local_v4(), local_v4())).unwrap() == LOOPBACK);
     }
 
     #[test]
-    fn loopback_interface_mtu_v6() {
-        #[cfg(target_os = "macos")]
-        check_mtu("localhost:443", false, 16384);
-        #[cfg(target_os = "linux")]
-        check_mtu("localhost:443", false, 65_536);
-        #[cfg(target_os = "windows")]
-        check_mtu("localhost:443", false, 4_294_967_295);
+    fn loopback_v4_loopback_v6() {
+        assert!(interface_and_mtu(&(local_v4(), local_v6())).is_err());
+    }
+
+    #[test]
+    fn loopback_v6_loopback_v4() {
+        assert!(interface_and_mtu(&(local_v6(), local_v4())).is_err());
+    }
+
+    #[test]
+    fn loopback_v6_loopback_v6() {
+        assert!(interface_and_mtu(&(local_v6(), local_v6())).unwrap() == LOOPBACK);
```

Comment:
```suggestion
        assert_eq!(interface_and_mtu(&(local_v6(), local_v6())).unwrap(), LOOPBACK);
```

...throughout.  The debugging is better if things break.